cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -102,17 +104,19 @@ func (c *Commander) watch() {
 
 	c.running.Store(0)
 	close(c.doneCh)
+	c.exitCh <- struct{}{}
+}
+
+// Exited returns a channel that will send a signal when the Agent process exits.
+func (c *Commander) Exited() <-chan struct{} {
+	return c.exitCh
 }
 
 // Done returns a channel that will send a signal when the Agent process is finished.
 func (c *Commander) Done() <-chan struct{} {
```

Comment:
I think we should remove this if we have `Exited`. I think it's fine to allow a single listener on `Exited` or to call `Stop`, which waits for this to fire anyway.

Reply:
I assumed the intent behind the `Done` is to indicate the program is successfully finished, as in something the supervisor will rely on before it cleans up all resources. Did I interpret that correctly? Otherwise, what would done channel be used for?

Reply:
That was my understanding of the purpose of `Done` as well, but in practice I don't think it's useful anymore. We need to essentially cover two situations that could happen with the Collector process:
1. The process exits unexpectedly.
2. We are shutting down and the process gracefully exits as a result of being asked to stop.

I think case 1 can be covered by the `Exit` channel, while case 2 is triggered by the `Stop` method, which blocks on `doneCh`. I don't see the need for the caller to wait on both `Stop` and `Done` since they both wait on the same channel. Also, `exitCh` and `doneCh` are fired at the same time anyway, so from outside of `Commander` they are functionally equivalent.